closed/make/Javadoc.gmk

```diff
@@ -0,0 +1,486 @@
+OPENJ9_MGMT_HEADER := <strong>OpenJ9 Monitoring and Management Interface Classes</strong>
+
+#############################################################################
+# The index.html, options, and packages files
+OPENJ9_MGMT_INDEXALL_FILE = $(OPENJ9_MGMT_DOCDIR)/index-all.html
+OPENJ9_MGMT_INDEX_FILE = $(OPENJ9_MGMT_DOCDIR)/index.html
+OPENJ9_MGMT_OPTIONS_FILE := $(DOCSTMPDIR)/openj9.mgmt.options
+OPENJ9_MGMT_PACKAGES_FILE := $(DOCSTMPDIR)/openj9.mgmt.packages
+
+#############################################################################
+# Set the link prefix for the JavaSE core and com.sun.management packages
+# point at the pages created in this build.
+JAVASE_BASE_URL := ../../$(JDKJRE2COREAPI)
+MGMT_BASE_URL := $(OPENJ9_MGMT_DOCDIR)
+OPENJ9_MGMT_SRCPATH := $(OPENJ9_DOCS_SRCDIR)$(PATH_SEP)$(TEMP_SUN_MGMT_PKGDIR)
+
+#############################################################################
+# Run javadoc if the index-all.html file is out of date or missing.
+# The openjdk build already uses index.html as its marker, so this
+# uses the index-all.html file instead.
+# The dependency on the index.html file ensures this javadoc
+# generation will replace the openjdk javadoc generation.
+$(OPENJ9_MGMT_INDEXALL_FILE): FIRST_COPYRIGHT_YEAR := $(OPENJ9_MGMT_FIRST_COPYRIGHT_YEAR)
+$(OPENJ9_MGMT_INDEXALL_FILE): $(OPENJ9_MGMT_INDEX_FILE) $(OPENJ9_MGMT_OPTIONS_FILE) $(OPENJ9_MGMT_PACKAGES_FILE)
+	$(MKDIR) -p $(TEMP_SUN_MGMT_SRCDIR)
+	if [ ! -d $(TEMP_SUN_MGMT_SRCDIR) ]; then echo Error: $(TEMP_SUN_MGMT_SRCDIR) does not exist, mkdir -p $(TEMP_SUN_MGMT_SRCDIR) must have failed; exit 1; fi
+	$(CP) $(SUN_MGMT_SRCDIR)/GarbageCollectorMXBean.java $(TEMP_SUN_MGMT_SRCDIR)
+	$(CP) $(SUN_MGMT_SRCDIR)/OperatingSystemMXBean.java $(TEMP_SUN_MGMT_SRCDIR)
+	$(CP) $(SUN_MGMT_SRCDIR)/UnixOperatingSystemMXBean.java $(TEMP_SUN_MGMT_SRCDIR)
+	$(CP) $(SUN_MGMT_SRCDIR)/package-info.java $(TEMP_SUN_MGMT_SRCDIR)
+	$(prep-javadoc)
+	$(call JavadocSummary,$(OPENJ9_MGMT_OPTIONS_FILE),$(OPENJ9_MGMT_PACKAGES_FILE))
+	$(JAVADOC_CMD) -d $(@D) @$(OPENJ9_MGMT_OPTIONS_FILE) @$(OPENJ9_MGMT_PACKAGES_FILE) 1>&2
+	$(RM) -r $(TEMP_SUN_MGMT_PKGDIR)
+
+$(OPENJ9_MGMT_OPTIONS_FILE):
+	$(prep-target)
+	@($(call OptionOnly,$(COMMON_JAVADOCFLAGS)) ; \
+	  $(call OptionPair,-sourcepath,$(OPENJ9_MGMT_SRCPATH)) ; \
+	  $(call OptionPair,-encoding,ascii) ; \
+	  $(call OptionPair,-doctitle,$(MGMT_DOCTITLE)) ; \
+	  $(call OptionPair,-windowtitle,$(MGMT_WINDOWTITLE) $(DRAFT_WINTITLE)) ; \
+	  $(call OptionPair,-header,$(MGMT_HEADER)$(DRAFT_HEADER)) ; \
+	  $(call OptionPair,-bottom,$(MGMT_BOTTOM)$(DRAFT_BOTTOM)) ; \
+	  $(call OptionTrip,-linkoffline,$(JAVASE_BASE_URL),$(COREAPI_DOCSDIR)/) ; \
+	  $(call OptionTrip,-linkoffline,$(MGMT_BASE_URL),$(MGMT_DOCDIR)/) ; \
+	) > $@
+
+# Create a file with the package names in it
+$(OPENJ9_MGMT_PACKAGES_FILE): $(DIRECTORY_CACHE) $(call PackageDependencies,$(ALL_MGMT_PKGS))
+	echo Running $@
```

Comment:
This should use `$(ECHO)` instead of `echo`.

Reply:
Actually the
```
echo Running $@
```
was a statement left in from debugging.  I'll remove it.